mediachain/transactor/blockchain_follower.py

```diff
@@ -65,48 +68,82 @@ def cancel(self):
         self._cancelled = True
         self.block_ref_queue.put('__abort__')
         self.incoming_event_queue.put('__abort__')
+        # setting these events will cause the __abort__ messages to be
+        # pulled from the queues.
+        self.catchup_begin.set()
+        self.catchup_complete.set()
+
+    def _clear_queues(self):
+        with self.block_ref_queue.mutex:
```

Comment:
hmm, that's a good point.  I can either wrap these with `foo_queue.mutex` or stop putting the `__abort__` sentinel in the queue and use a `threading.Event` to signal cancellation instead.  I think I like the event better, since it makes the queue single-purpose instead of conflating these two things.